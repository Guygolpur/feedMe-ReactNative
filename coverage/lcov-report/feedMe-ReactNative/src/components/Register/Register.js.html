<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for feedMe-ReactNative/src/components/Register/Register.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">feedMe-ReactNative/src/components/Register</a> Register.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/18</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a>
<a name='L109'></a><a href='#L109'>109</a>
<a name='L110'></a><a href='#L110'>110</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import { Image, Button, StyleSheet, Text, View } from 'react-native'
import * as Expo from 'expo'
import { androidClientId, iosClientId } from '../../../superSecretKey'
import propTypes from 'prop-types'
&nbsp;
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center'
  },
  header: {
    fontSize: 25
  },
  image: {
    marginTop: 15,
    width: 150,
    height: 150,
    borderColor: 'rgba(0,0,0,0.2)',
    borderWidth: 3,
    borderRadius: 150
  },
  continue: {
    marginTop: 25
  }
})
&nbsp;
// export let Gmail = null
export default class Register extends Component {
  constructor(props) {
    super(props)
    this.state = {
      signedIn: false,
      name: '',
      photoUrl: '',
      email: ''
    }
    this.LoggedInPage = this.LoggedInPage.bind(this)
    this.signIn = this.signIn.bind(this)
    this.LoginPage = this.LoginPage.bind(this)
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  Lo</span>ggedInPage(props) {
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;View style={styles.container}&gt;
        &lt;Text style={styles.header}&gt;Welcome:{props.name}&lt;/Text&gt;
        &lt;Image style={styles.image} source={{ uri: props.photoUrl }} /&gt;
        &lt;View style={styles.continue}&gt;
          &lt;Button onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.props.changeGmailMode()}</span> title="Continue"&gt;&lt;/Button&gt;
        &lt;/View&gt;
      &lt;/View&gt;
    )
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  as</span>ync signIn() {
    const result = <span class="cstat-no" title="statement not covered" >await Expo.Google.logInAsync({</span>
      androidClientId,
      iosClientId,
      scopes: ['profile', 'email']
    })
<span class="cstat-no" title="statement not covered" >    if (result.type === 'success') {</span>
<span class="cstat-no" title="statement not covered" >      this.setState(</span>
        {
          signedIn: true,
          name: result.user.name,
          photoUrl: result.user.photoUrl,
          email: result.user.email
        },
<span class="fstat-no" title="function not covered" >        ()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >          this.props.changeGmailAccount(this.state.email)</span>
          const url = <span class="cstat-no" title="statement not covered" >`https://feedme24.herokuapp.com/addProfile`</span>
<span class="cstat-no" title="statement not covered" >          fetch(`${url}`, {</span>
            method: 'POST',
            body: `gmailAccount=${this.state.email}`,
            headers: {
              'Content-Type': 'application/x-www-form-urlencoded'
            }
          })
        }
      )
    }
  }
  LoginPage(props) {
    return (
      &lt;View&gt;
        &lt;Text style={styles.header}&gt;Sign In With Google&lt;/Text&gt;
        &lt;Button title="Sign in with Google" onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >props.signIn()}</span> /&gt;
      &lt;/View&gt;
    )
  }
  render() {
    return (
      &lt;View style={styles.container}&gt;
        {this.state.signedIn ? (
<span class="branch-0 cbranch-no" title="branch not covered" >          &lt;this.LoggedInPage name={this.state.name} photoUrl={this.state.photoUrl} /&gt;</span>
        ) : (
          &lt;this.LoginPage signIn={this.signIn} /&gt;
        )}
      &lt;/View&gt;
    )
  }
}
&nbsp;
Register.propTypes = {
  changeGmailMode: propTypes.func,
  changeGmailAccount: propTypes.func
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jun 28 2019 17:16:16 GMT+0300 (Israel Daylight Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
<script src="../../../../block-navigation.js"></script>
</body>
</html>
