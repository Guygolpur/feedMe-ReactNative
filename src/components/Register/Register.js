import React, { Component } from 'react'
import { Image, Button, StyleSheet, Text, View } from 'react-native'
import * as Expo from 'expo'
import { androidClientId, iosClientId } from '../../../superSecretKey'
import propTypes from 'prop-types'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center'
  },
  header: {
    fontSize: 25
  },
  image: {
    marginTop: 15,
    width: 150,
    height: 150,
    borderColor: 'rgba(0,0,0,0.2)',
    borderWidth: 3,
    borderRadius: 150
  },
  continue: {
    marginTop: 25
  }
})

// export let Gmail = null
export default class Register extends Component {
  constructor(props) {
    super(props)
    this.state = {
      signedIn: false,
      name: '',
      photoUrl: '',
      email: ''
    }
    this.LoggedInPage = this.LoggedInPage.bind(this)
    this.signIn = this.signIn.bind(this)
    this.LoginPage = this.LoginPage.bind(this)
  }

  LoggedInPage(props) {
    return (
      <View style={styles.container}>
        <Text style={styles.header}>Welcome:{props.name}</Text>
        <Image style={styles.image} source={{ uri: props.photoUrl }} />
        <View style={styles.continue}>
          <Button onPress={() => this.props.changeGmailMode()} title="Continue"></Button>
        </View>
      </View>
    )
  }

  async signIn() {
    const result = await Expo.Google.logInAsync({
      androidClientId,
      iosClientId,
      scopes: ['profile', 'email']
    })
    if (result.type === 'success') {
      this.setState(
        {
          signedIn: true,
          name: result.user.name,
          photoUrl: result.user.photoUrl,
          email: result.user.email
        },
        () => {
          this.props.changeGmailAccount(this.state.email)
          const url = `https://feedme24.herokuapp.com/addProfile`
          fetch(`${url}`, {
            method: 'POST',
            body: `gmailAccount=${this.state.email}`,
            headers: {
              'Content-Type': 'application/x-www-form-urlencoded'
            }
          })
        }
      )
    }
  }
  LoginPage(props) {
    return (
      <View>
        <Text style={styles.header}>Sign In With Google</Text>
        <Button title="Sign in with Google" onPress={() => props.signIn()} />
      </View>
    )
  }
  render() {
    return (
      <View style={styles.container}>
        {this.state.signedIn ? (
          <this.LoggedInPage name={this.state.name} photoUrl={this.state.photoUrl} />
        ) : (
          <this.LoginPage signIn={this.signIn} />
        )}
      </View>
    )
  }
}

Register.propTypes = {
  changeGmailMode: propTypes.func,
  changeGmailAccount: propTypes.func
}
