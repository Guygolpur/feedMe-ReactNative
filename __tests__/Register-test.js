import React from 'react'
import renderer from 'react-test-renderer'
import Register from '../src/components/Register/Register'

if (!global.fetch) {
  global.fetch = require('node-fetch')
}

jest.mock('expo', () => ({
  Permissions: {
    askAsync: jest.fn()
  }
}))

describe('Register component tests', () => {
  test('renders correctly', async () => {
    const mockFunck = jest.fn()
    const tree = await renderer
      .create(<Register changeGmailMode={mockFunck} changeGmailAccount={mockFunck} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('check props exist', async () => {
    const changeGmailMode = jest.fn()
    const changeGmailAccount = jest.fn()
    const testRenderer = await renderer.create(
      <Register changeGmailMode={changeGmailMode} changeGmailAccount={changeGmailAccount} />
    )
    expect(testRenderer.root.props).toHaveProperty('changeGmailMode')
    expect(testRenderer.root.props).toHaveProperty('changeGmailAccount')
  })
  test('call action changeGmailMode', async () => {
    const changeGmailMode = jest.fn()
    const changeGmailAccount = jest.fn()
    const testRenderer = renderer.create(
      <Register changeGmailMode={changeGmailMode} changeGmailAccount={changeGmailAccount} />
    )
    const RegisterComponent = testRenderer.root.findAllByType(Register)[0]
    RegisterComponent.props.changeGmailMode()
    expect(changeGmailMode).toHaveBeenCalledTimes(1)
    RegisterComponent.props.changeGmailAccount('barazouri24@gmail.com')
    expect(changeGmailAccount).toHaveBeenCalledTimes(1)
  })
})
