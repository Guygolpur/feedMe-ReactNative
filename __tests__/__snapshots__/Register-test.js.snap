// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Sign In With Google
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign in with Google
        </Text>
      </View>
    </View>
  </View>
</View>
`;
